src/sentry/spans/consumers/process/flusher.py

```
@@ -253,15 +326,22 @@ def close(self) -> None:
-        # set stopped flag first so we can "flush" the background thread while
+        # set stopped flag first so we can "flush" the background threads while
         # next_step is also shutting down. we can do two things at once!
         self.stopped.value = True
         deadline = time.time() + timeout if timeout else None
 
         self.next_step.join(timeout)
 
-        while self.process.is_alive() and (deadline is None or deadline > time.time()):
-            time.sleep(0.1)
+        # Wait for all processes to finish
+        for process_index, process in self.processes.items():
```

Ruff (0.16.2)

[warning] Loop control variable `process_index` not used within loop body

Rename unused `process_index` to `_process_index`

(B007)